pkg/intel/metadata/bg/bgkey/manifest_manifestcodegen.go

```diff
@@ -0,0 +1,336 @@
+		}
+		totalN += int64(n)
+	}
+
+	// BPKey (ManifestFieldType: subStruct)
+	{
+		n, err := s.BPKey.WriteTo(w)
+		if err != nil {
+			return totalN, fmt.Errorf("unable to write field 'BPKey': %w", err)
+		}
+		totalN += int64(n)
+	}
+
+	// KeyAndSignature (ManifestFieldType: subStruct)
+	{
+		n, err := s.KeyAndSignature.WriteTo(w)
+		if err != nil {
+			return totalN, fmt.Errorf("unable to write field 'KeyAndSignature': %w", err)
+		}
+		totalN += int64(n)
+	}
+
+	return totalN, nil
+}
+
+// StructInfoSize returns the size in bytes of the value of field StructInfo
+func (s *Manifest) StructInfoTotalSize() uint64 {
+	return s.StructInfo.TotalSize()
+}
+
+// KMVersionSize returns the size in bytes of the value of field KMVersion
+func (s *Manifest) KMVersionTotalSize() uint64 {
+	return 1
+}
+
+// KMSVNSize returns the size in bytes of the value of field KMSVN
+func (s *Manifest) KMSVNTotalSize() uint64 {
+	return 1
+}
+
+// KMIDSize returns the size in bytes of the value of field KMID
+func (s *Manifest) KMIDTotalSize() uint64 {
+	return 1
+}
+
+// BPKeySize returns the size in bytes of the value of field BPKey
+func (s *Manifest) BPKeyTotalSize() uint64 {
+	return s.BPKey.TotalSize()
+}
+
+// KeyAndSignatureSize returns the size in bytes of the value of field KeyAndSignature
+func (s *Manifest) KeyAndSignatureTotalSize() uint64 {
+	return s.KeyAndSignature.TotalSize()
+}
+
+// StructInfoOffset returns the offset in bytes of field StructInfo
+func (s *Manifest) StructInfoOffset() uint64 {
+	return 0
+}
+
+// KMVersionOffset returns the offset in bytes of field KMVersion
+func (s *Manifest) KMVersionOffset() uint64 {
+	return s.StructInfoOffset() + s.StructInfoTotalSize()
+}
+
+// KMSVNOffset returns the offset in bytes of field KMSVN
+func (s *Manifest) KMSVNOffset() uint64 {
+	return s.KMVersionOffset() + s.KMVersionTotalSize()
+}
+
+// KMIDOffset returns the offset in bytes of field KMID
+func (s *Manifest) KMIDOffset() uint64 {
+	return s.KMSVNOffset() + s.KMSVNTotalSize()
+}
+
+// BPKeyOffset returns the offset in bytes of field BPKey
+func (s *Manifest) BPKeyOffset() uint64 {
+	return s.KMIDOffset() + s.KMIDTotalSize()
+}
+
+// KeyAndSignatureOffset returns the offset in bytes of field KeyAndSignature
+func (s *Manifest) KeyAndSignatureOffset() uint64 {
+	return s.BPKeyOffset() + s.BPKeyTotalSize()
+}
+
+// Size returns the total size of the Manifest.
+func (s *Manifest) TotalSize() uint64 {
+	if s == nil {
+		return 0
+	}
+
+	var size uint64
+	size += s.StructInfoTotalSize()
+	size += s.KMVersionTotalSize()
+	size += s.KMSVNTotalSize()
+	size += s.KMIDTotalSize()
+	size += s.BPKeyTotalSize()
+	size += s.KeyAndSignatureTotalSize()
+	return size
+}
+
+// PrettyString returns the content of the structure in an easy-to-read format.
+func (s *Manifest) PrettyString(depth uint, withHeader bool, opts ...pretty.Option) string {
+	var lines []string
+	if withHeader {
+		lines = append(lines, pretty.Header(depth, "Manifest", s))
```

Comment:
proposal: change this into `"Key Manifest"` to make it easier to spot right away
in a similar fashion, the BPM prints as `"Boot Policy Manifest"`

Reply:
https://github.com/linuxboot/fiano/pull/396

take it or leave it :)